RPP - cleanup and updates (#584)
GPUOpen-ProfessionalCompute-Libraries/rpp: docs/install/rpp-prerequisites.rst

The following compilers and libraries are required to build and install RPP:

* HIP
* OpenMP
* half, the half-precision floating-point library, version 1.12.0 and above
* libstdc++-12-dev for Ubuntu 22.04 only
* Clang version 5.0.1 or later for CPU-only backends

> * AMD Clang++ Version 18.0.0 and above for HIP and OpenCL backends

Reviewer: ```suggestion
* AMD Clang++ Version 18.0.0 or later for HIP and OpenCL backends

```

The following compilers and libraries are required to build and install RPP:

* HIP
* OpenMP
* half, the half-precision floating-point library, version 1.12.0 and above
* libstdc++-12-dev for Ubuntu 22.04 only

> * Clang version 5.0.1 and above for CPU-only backends

Reviewer: ```suggestion
* Clang version 5.0.1 or later for CPU-only backends
```

This is a style guide issue. The ROCm style guide specifies later rather than above.

* AMD Clang++ Version 18.0.0 and above for HIP and OpenCL backends

With the following compiler support:

> * C++17 and above

Reviewer: ```suggestion
* C++17 or later
```
